doc(2.1.x) update brain-immunity topics (#2236)
Kong/docs.konghq.com: app/enterprise/2.1.x/brain-immunity/overview.md

---
title: Kong Brain & Kong Immunity Overview
---

**Kong Brain** and **Kong Immunity** help automate the entire API and service development life cycle. By automating processes for configuration, traffic analysis and the creation of documentation, Kong Brain and Kong Immunity help organizations improve efficiency, governance, reliability and security. Kong Brain automates API and service documentation and Kong Immunity uses advanced machine learning to analyze traffic patterns to diagnose and improve security.

![Kong Brain and Kong Immunity](/assets/images/docs/ee/brain_immunity_overview.png)

### About Kong Brain

> Kong Brain automates tasks, increases visibility and ensures consistency across complex architectures. Other features include:

Reviewer: ```suggestion
Kong Brain automates tasks, increases visibility, and ensures consistency across complex architectures. Other features include:
```

#### Visually Map Services
Get a high-level view of your architecture with a real-time visual service map. Analyze inter-service dependencies across teams and platforms to improve governance and minimize risk. See [Using the Service Map](/enterprise/latest/brain-immunity/service-map/).

#### Auto-generate Documentation
Standardize docs with hands-free creation of OpenAPI spec files for all your services. Transition seamlessly to spec-first development to increase collaboration and developer efficiency. See [Auto-Generated Specs](/enterprise/latest/brain-immunity/auto-generated-specs/).

#### Streamline Issue Resolution
Contextualize issues and take immediate action by surfacing Kong Immunity anomaly alerts in the service map. Rapidly identify issues impacting services to reduce risk and minimize downtime.

### About Kong Immunity
Kong Immunity autonomously identifies service issues with machine learning-powered anomaly detection. Other features include:

#### Create a Baseline for Healthy Traffic

> Traffic patterns provide a window into the behavior and performance of services under different conditions. To understand your existing traffic patterns, Kong Immunity ingests data flowing through the Kong data plane to create a baseline for healthy traffic. As anomalies are detected and addressed through changes to the Kong Enterprise configuration, Kong Immunity continuously adapts this baseline. Kong Immunity uses the Median Average Deviation (MAD) of traffic to detect anomalies. The dataset used to calculate the expected MAD continuously updates based on the last week of data available.

Reviewer 2: We only use MAD for traffic count anomalies.  If you need to specify some more how the other alerts are generated, most of it are statistical methods (including MAD).  But I would also talk to marketing/sales first about that because they have been strongly insinuating that Machine Learning is behind Immunity, and if you say "statistical methods" are what power Immunity, then the strong insinuation of Machine Learning is debunked immediately.

#### Autonomously Identify Anomalies

> To identify potential issues, inefficiencies or performance bottlenecks, Kong Immunity flags traffic that deviates from the expected or desired patterns without disrupting services. Depending on your needs and goals, you can adjust the settings of Kong Immunity to recognize individual traffic events, patterns and other types of anomalous activity.

Reviewer: ```suggestion
To identify potential issues, inefficiencies or performance bottlenecks, Kong Immunity flags traffic that deviates from the expected or desired patterns without disrupting services. Depending on your requirements, you can adjust the settings of Kong Immunity to recognize individual traffic events, patterns, and other types of anomalous activity.
```

#### Automatically Alert

> How quickly you respond to an event can mean the difference between a simple fix and catastrophic outage. As Kong Immunity detects anomalies in real-time, it automatically sends a notification alerting you to the issue. Notifications are available via email, Slack and more. To avoid disruptions to your teams, you can designate specific users to receive alerts based on Roles Based Access Controls (RBAC) within Kong Manager. 

Reviewer: ```suggestion
How quickly you respond to an event can mean the difference between a simple fix and a catastrophic outage. As Kong Immunity detects anomalies in real-time, it automatically sends a notification alerting you to the issue. Notifications are available via email, Slack, and more. To avoid disruptions to your teams, you can designate specific users to receive alerts based on Role-Based Access Controls (RBAC) within Kong Manager. 
```
